src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2067,8 +2075,16 @@ internal static bool CheckValEscape(SyntaxNode node, BoundExpression expr, uint
                 case BoundKind.ConditionalOperator:
                     var conditional = (BoundConditionalOperator)expr;
 
-                    return CheckValEscape(conditional.Consequence.Syntax, conditional.Consequence, escapeFrom, escapeTo, checkingReceiver: false, diagnostics: diagnostics) &&
-                           CheckValEscape(conditional.Alternative.Syntax, conditional.Alternative, escapeFrom, escapeTo, checkingReceiver: false, diagnostics: diagnostics);
+                    var consValid = CheckValEscape(conditional.Consequence.Syntax, conditional.Consequence, escapeFrom, escapeTo, checkingReceiver: false, diagnostics: diagnostics);
+
+                    if (consValid || conditional.IsByRef)
+                    {
```

Comment:
Not sure I get that. Why don't we check the alternative if the consequence is valid?

Reply:
Should be !consValid. And I need test, I guess.

---
In reply to: [139840188](https://github.com/dotnet/roslyn/pull/22203#discussion_r139840188) [](ancestors = 139840188)